### ISSUE TITLE
Crashed: com.google.firebase.firestore SIGABRT ABORT 0x00000001b21a7414 0

### ISSUE THREAD

**Issue by sspogra** [firebase/firebase-ios-sdk#7073]
### Step 0: Are you in the right place?

  * For issues or feature requests related to __the code in this repository__
    file a Github issue.
    * If this is a __feature request__ please use the Feature Request template.
  * For general technical questions, post a question on [StackOverflow](http://stackoverflow.com/)
    with the `firebase` tag.
  * For general (non-iOS) Firebase discussion, use the [firebase-talk](https://groups.google.com/forum/#!forum/firebase-talk)
    google group.
  * For backend issues, console issues, and other non-SDK help that does not fall under one
    of the above categories, reach out to
    [Firebase Support](https://firebase.google.com/support/).
  * Once you've read this section and determined that your issue is appropriate for
    this repository, please delete this section.

### [REQUIRED] Step 1: Describe your environment

  * Xcode version:12.0.1
  * Firebase SDK version:7.0.0
  * Installation method: `CocoaPods
  * Firebase Component: Auth, Core, Database, Firestore, Messaging

### [REQUIRED] Step 2: Describe the problem

```
Crashed: com.google.firebase.firestore
SIGABRT ABORT 0x00000001b21a7414
0
libsystem_kernel.dylib
__pthread_kill + 8
1
libsystem_pthread.dylib
pthread_kill + 272
2
libsystem_c.dylib
abort + 104
3
libsystem_malloc.dylib
_malloc_put + 558
4
libsystem_malloc.dylib
malloc_report + 64
5
libsystem_malloc.dylib
free + 532
6
FirebaseFirestore
firebase::firestore::immutable::SortedMap<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, firebase::firestore::model::FieldValue, firebase::firestore::util::Comparator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > >::operator=(firebase::firestore::immutable::SortedMap<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, firebase::firestore::model::FieldValue, firebase::firestore::util::Comparator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > >&&) + 204
7
FirebaseFirestore
firebase::firestore::remote::Serializer::DecodeFields(firebase::firestore::nanopb::Reader*, unsigned long, firebase::firestore::_google_firestore_v1_Document_FieldsEntry const*) const + 152
8
FirebaseFirestore
firebase::firestore::remote::Serializer::DecodeMutation(firebase::firestore::nanopb::Reader*, firebase::firestore::_google_firestore_v1_Write const&) const + 188
9
FirebaseFirestore
firebase::firestore::local::LocalSerializer::DecodeMutationBatch(firebase::firestore::nanopb::Reader*, firebase::firestore::_firestore_client_WriteBatch const&) const + 244
10
FirebaseFirestore
firebase::firestore::local::LevelDbMutationQueue::ParseMutationBatch(absl::lts_2020_02_25::string_view) + 84
11
FirebaseFirestore
firebase::firestore::local::LevelDbMutationQueue::NextMutationBatchAfterBatchId(int) + 444
12
FirebaseFirestore
std::__1::__function::__func<std::__1::enable_if<!(std::is_same<void, decltype(fp0())>::value), decltype(fp0())>::type firebase::firestore::local::Persistence::Run<firebase::firestore::local::LocalStore::GetNextMutationBatch(int)::$_9>(absl::lts_2020_02_25::string_view, firebase::firestore::local::LocalStore::GetNextMutationBatch(int)::$_9)::'lambda'(), std::__1::allocator<std::__1::enable_if<!(std::is_same<void, decltype(fp0())>::value), decltype(fp0())>::type firebase::firestore::local::Persistence::Run<firebase::firestore::local::LocalStore::GetNextMutationBatch(int)::$_9>(absl::lts_2020_02_25::string_view, firebase::firestore::local::LocalStore::GetNextMutationBatch(int)::$_9)::'lambda'()>, void ()>::operator()() + 52
13
FirebaseFirestore
firebase::firestore::local::LevelDbPersistence::RunInternal(absl::lts_2020_02_25::string_view, std::__1::function<void ()>) + 136
14
FirebaseFirestore
firebase::firestore::local::LocalStore::GetNextMutationBatch(int) + 152
15
FirebaseFirestore
firebase::firestore::remote::RemoteStore::FillWritePipeline() + 116
16
FirebaseFirestore
firebase::firestore::core::SyncEngine::WriteMutations(std::__1::vector<firebase::firestore::model::Mutation, std::__1::allocator<firebase::firestore::model::Mutation> >&&, std::__1::function<void (firebase::firestore::util::Status)>) + 828
17
FirebaseFirestore
std::__1::__function::__func<firebase::firestore::core::FirestoreClient::WriteMutations(std::__1::vector<firebase::firestore::model::Mutation, std::__1::allocator<firebase::firestore::model::Mutation> >&&, std::__1::function<void (firebase::firestore::util::Status)>)::$_14, std::__1::allocator<firebase::firestore::core::FirestoreClient::WriteMutations(std::__1::vector<firebase::firestore::model::Mutation, std::__1::allocator<firebase::firestore::model::Mutation> >&&, std::__1::function<void (firebase::firestore::util::Status)>)::$_14>, void ()>::operator()() + 152
18
FirebaseFirestore
firebase::firestore::util::AsyncQueue::ExecuteBlocking(std::__1::function<void ()> const&) + 68
19
FirebaseFirestore
firebase::firestore::util::Task::ExecuteAndRelease() + 184
20
libdispatch.dylib
_dispatch_client_callout + 20
21
libdispatch.dylib
_dispatch_lane_serial_drain + 580
22
libdispatch.dylib
_dispatch_lane_invoke + 408
23
libdispatch.dylib
_dispatch_workloop_worker_thread + 708
24
libsystem_pthread.dylib
_pthread_wqthread + 276
25
libsystem_pthread.dylib
start_wqthread + 8
```

#### Steps to reproduce:

App is 100% in background and crash reported more then 13 times.

#### Relevant Code:

```
```

**Comment by paulb777** [comment]
Perhaps related to #6844. Updating to 7.1.0 may help.

**Comment by wilhuff** [comment]
This was fixed in #6849 and that fix was released with Firebase 7.1.0.

Note that this is actually harmless. The crash only happens when a user force closes the app. It happens to be the case that the app is crashing instead of exiting cleanly but this isn't something the user can tell and regardless, the user wanted the app gone and now it's gone :-).
